drivers/ata8510/ata8510.c

```diff
@@ -0,0 +1,543 @@
+
+static char stack[THREAD_STACKSIZE_DEFAULT];
+static void *_recv_thread(void *arg);
+
+static void _irq_handler(void *arg)
+{
+    ata8510_t *dev = (ata8510_t *)arg;
+
+    switch (dev->internal_state) {
+        case ATA8510_MODE_IDLE:
+            DEBUG("[ata8510] _irq_handler: event received in mode IDLE\n");
+            break;
+        case ATA8510_MODE_TX:
+            DEBUG("[ata8510] _irq_handler: event received in mode TX\n");
+            mutex_unlock(&(dev->tx_lock));
+            break;
+        case ATA8510_MODE_RX:
+            DEBUG("[ata8510] _irq_handler: event received in mode RX\n");
+            mutex_unlock(&(dev->rx_lock));
+            break;
+        case ATA8510_MODE_RX_POLLING:
+            DEBUG(
+                "[ata8510] _irq_handler: event received in mode RX POLLING\n"
+                );
+            mutex_unlock(&(dev->rx_lock));
+            break;
+    }
+}
+
+static void _send_command(ata8510_t *dev, const void *out,
+                          void *in, size_t len)
+{
+    if (spi_acquire(SPIDEV, CSPIN, SPI_MODE_0, dev->params.spi_clk) < 0) {
+        DEBUG("[ata8510] ERROR: Cannot acquire SPI bus!\n");
+    }
+    gpio_clear(CSPIN);
+    xtimer_usleep(1);
+    spi_transfer_bytes(SPIDEV, SPI_CS_UNDEF, true, out, in, len);
+    xtimer_usleep(1);
+    gpio_set(CSPIN);
+    spi_release(SPIDEV);
+}
+
+int ata8510_init(ata8510_t *dev, const ata8510_params_t *params)
+{
+    int res;
+
+    // write config params to device descriptor
+    dev->params = *params;
+    dev->receive_cb = NULL;
+
+    // Initialize pins
+    if (gpio_init_int(INTPIN, GPIO_IN,
+                      GPIO_FALLING, _irq_handler, dev) < 0) {
+        DEBUG("[ata8510] ERROR: Interrupt pin not initialized\n");
+        return -ATA8510_ERR_GPIO_INT;
+    }
+    if (gpio_init(SLEEPPIN, GPIO_OUT) < 0) {
+        DEBUG("[ata8510] ERROR: Power pin not initialized\n");
+        return -ATA8510_ERR_GPIO_POWER;
+    }
+    if (gpio_init(RESETPIN, GPIO_OUT) < 0) {
+        DEBUG("[ata8510] ERROR: Reset pin not initialized\n");
+        return -ATA8510_ERR_GPIO_RESET;
+    }
+
+    ata8510_poweron(dev);
+
+    // Initialize SPI bus
+    if (spi_init_cs(dev->params.spi, CSPIN) < 0) {
+        DEBUG("[ata8510] ERROR: SPI not initialized\n");
+        return -ATA8510_ERR_SPI;
+    }
+
+    uint8_t sig = ata8510_get_device_signature(dev);
+    if (sig != ATA8510_PARTNUM) {
+        DEBUG("[ata8510] ERROR: signature is not 0x%02x.\n", ATA8510_PARTNUM);
+        return -ATA8510_ERR_SIGNATURE;
+    }
+
+    // Clear event line
+    ata8510_get_event_bytes(dev);
+
+    dev->channel = ATA8510_CHANNEL_0;
+    dev->service = ATA8510_SERVICE_0;
+
+    // _recv_thread will block on rx_lock until a frame arrives
+    mutex_lock(&(dev->rx_lock));
+    res = thread_create(stack, sizeof(stack), THREAD_PRIORITY_MAIN - 1,
+                        THREAD_CREATE_STACKTEST, _recv_thread, (void *)dev,
+                        "recv_thread");
```

Comment:
See above - this is still not a full netdev. Would it make sense to simply export a new _ata8510_recv_thread_ function, and use that in the test for the time being?
It could be easily hooked into auto_init as soon as I implement the remaining stuff.